### PR TITLE
Fix issue 22137 - -preview=dip1000 enables visibility checks for tupleof

### PR DESCRIPTION
**Pull request by dkorpel**
Reverts https://github.com/dlang/dmd/pull/8035 and https://github.com/dlang/dmd/pull/12545

This assumes the future of `@safe` encapsulation is `@system` variables, instead of increasing the scope of `@trusted` from function-level to module-level.

Also fixes [Issue 19877 - [dip1000] std.container.rbtree is unsafely accessing private data](https://issues.dlang.org/show_bug.cgi?id=19877).

### CODE REVIEW

File: test/fail_compilation/test18554.d [hunk 1]
```diff
@@ -1,24 +0,0 @@
-/* REQUIRED_ARGS: -preview=dip1000
-EXTRA_FILES: imports/imp18554.d
```
Comment (MoonlightSentinel) [inline test/fail_compilation/test18554.d:2]: Why is this test removed instead of changing it to match the new/old behavior?

**Comment by RazvanN7** [comment]
Pasting @WalterBright 's comment from the PR that is reverted:

```
A struct may present an @safe interface to users, but internally have private unsafe pointers. tupleof allows any code to have access to those private members, and can mess up the invariants relied on by the struct. Even int fields are not safe to access, as they may be used as an index to a pointer.

If you're going to access private members in another module, bypassing the safe interface provided by that module, you're in @System land.
```

This seems like a sensible argument, what is your argument against this?

**Comment by 12345swordy** [comment]
@RazvanN7 I believe that this PR is depending on this Dip being accepted.
https://github.com/dlang/DIPs/blob/master/DIPs/DIP1035.md

File: test/fail_compilation/test18554.d (hunk 1 above)
Comment (dkorpel) [reply test/fail_compilation/test18554.d:2]: The tests were for code that isn't there anymore. I didn't think it made sense to test the absence of a special case, it's not something you would accidentally write again.

Comment (MoonlightSentinel) [reply test/fail_compilation/test18554.d:2]: True. But a quick search didn't yield any other test for visibility checks w.r.t. `tupleof` - might be good to either adapt it or integrate it into another test

**Review by RazvanN7 — APPROVED** [review]

**Pull request merged**